tests/phpunit/CRM/Contribute/Form/ContributionTest.php

```diff
@@ -859,19 +859,26 @@ public function testSubmitWithOutSaleTax() {
   public function testReSubmitSaleTax() {
     $this->enableTaxAndInvoicing();
     $this->relationForFinancialTypeWithFinancialAccount($this->_financialTypeId);
-    $form = new CRM_Contribute_Form_Contribution();
```

Comment:
Hmm. I can't find doInitialSubmit when grepping current master:
`Joes-MacBook-Pro-2:civicrm-core-master josephmurray$ grep -R doInitialSubmit ./*`
I'm wondering if this doInitialSubmit also engages the payment processor code. We've generally only ever worried about the contribution BAO, and have tried to stay away from everything to do with payment processors for separation of concerns reasons. I know you've done so much continuing work on payment processors, so I'm interested in understanding more about why we should be testing across both payment processor and contribution code, if that is the point of doInitialSubmit.